src/NBitcoin/Consensus.cs

```diff
@@ -83,12 +83,21 @@ public class Consensus : IConsensus
         public ConsensusFactory ConsensusFactory { get; }
 
         /// <inheritdoc />
-        public ICollection<IConsensusRule> Rules { get; set; }
+        public List<IIntegrityValidationConsensusRule> IntegrityValidationRules { get; set; }
+
+        /// <inheritdoc />
+        public List<IHeaderValidationConsensusRule> HeaderValidationRules { get; set; }
```

Comment:
The plan was to create a new class called ConsensusRules (like ConsensusFactory) and put the lists in that class.
So you would call  network.Consensus.Rules.HeaderValidationRules